auth.py

```
@@ -0,0 +1,200 @@
+                                    return self.session_id
+                    
+                    # Check response body for errors
+                    if login_response.status == 200:
+                        response_text = await login_response.text()
+                        if "incorrect" in response_text.lower() or "invalid" in response_text.lower():
+                            _LOGGER.error("Invalid Device Access Code")
+                        else:
+                            _LOGGER.error("Login failed - no session cookie received")
+                    
+        except asyncio.TimeoutError:
+            _LOGGER.error("Login timeout - router not responding")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)